# Data Exploration with Python using Pandas & Numpy Libraries

Group Number 8 : Itighne Sinha (055015) | Suvra Dutta Banik (055049)

Project Title | Student Name(s) | Enrollment Number(s) | Group Number(s)

## Decription Of Data

-   Source: The dataset has been sourced using Kaggle, titled 'Import Export 15000' by Vishwas Chakilam.
-   Type : Panel


In [32]:
#The required Imports
import pandas as pd
import numpy as np

In [34]:
#Loading the csv Data in Python
database=pd.read_csv("C:/Users/itigh/Imports_Exports_Dataset.csv")

In [47]:
# Viewing the top 5 rows of the data
database.head(5)

Transaction_ID           Country    Product  \
0  e3e70682-c209-4cac-a29f-6fbed82c07cd          Colombia   describe   
1  f728b4fa-4248-4e3a-8a5d-2f346baa9455             Chile  president   
2  eb1167b3-67a9-4378-bc65-c1e582e2e662            Turkey        far   
3  f7c1bd87-4da5-4709-9471-3d60c8a70639  Christmas Island     agency   
4  e443df78-9558-467f-9ba9-1faf7a024204           Finland     policy   

  Import_Export  Quantity    Value        Date     Category              Port  \
0        Export      1979  9506.57  07-12-2023    Machinery        Robertbury   
1        Export      5763  7100.91  04-04-2023     Clothing          Shahport   
2        Import      5740  2450.34  21-08-2024  Electronics  South Joshuatown   
3        Export      2592  7226.42  09-05-2022    Furniture          Adamfort   
4        Export      2622  2817.29  03-04-2023  Electronics         Juliebury   

   Customs_Code   Weight Shipping_Method                         Supplier  \
0        620537  4248.65             Air  Garrison, Hubbard and Hendricks   
1        927600  4340.81             Air                        Webb-Mack   
2        299258  4538.41             Air        Mendez, Jones and Johnson   
3        996084  4886.14             Air                  Schroeder-Smith   
4        555981  4406.74             Air                    Zimmerman LLC   

        Customer  Invoice_Number     Payment_Terms  
0      Seth Hall        21000294  Cash on Delivery  
1  Kimberly Ryan        88738432           Prepaid  
2     Ryan Silva        89922099           Prepaid  
3     Jacob Gray        63216265            Net 60  
4   Amy Stephens        94600248  Cash on Delivery

In [42]:
# Dimensions of the Data
database.shape

(15000, 16)

The dataset has 15000 observations and 16 variables

In [53]:
#Data Variable Types
#Viewing the information about the columns(Object Types) of the Data set
database.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15000 entries, 0 to 14999
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Transaction_ID   15000 non-null  object 
 1   Country          15000 non-null  object 
 2   Product          15000 non-null  object 
 3   Import_Export    15000 non-null  object 
 4   Quantity         15000 non-null  int64  
 5   Value            15000 non-null  float64
 6   Date             15000 non-null  object 
 7   Category         15000 non-null  object 
 8   Port             15000 non-null  object 
 9   Customs_Code     15000 non-null  int64  
 10  Weight           15000 non-null  float64
 11  Shipping_Method  15000 non-null  object 
 12  Supplier         15000 non-null  object 
 13  Customer         15000 non-null  object 
 14  Invoice_Number   15000 non-null  int64  
 15  Payment_Terms    15000 non-null  object 
dtypes: float64(2), int64(3), object(11)
memory usage: 1.8+ MB


Data Variable Category
- Index_variable = ['Transaction_ID', 'Date','Invoice_Number','Customs Code']
- Non_categorical_variable = ['Quantity', 'Value', 'Weight']
- Categorical_nominal_variable = ['Country', 'Product', 'Shipping_Method', 'Supplier', 'Customer', 'Import_Export','Port', 'Category']
- Categorical_ordinal_variable = ['Payment_Terms']

### Sampling of Data

In [134]:
# Sampled 2001 rows randomly from the given data set
samData = database.sample(n=2001,random_state=50155049)

In [136]:
# Display the sampled dataset
samData

Transaction_ID  \
4475   040c4d58-954a-4019-a093-e30f1e1eedd9   
14872  a171d7ab-9aaf-46b5-9885-75e4cbb1a283   
11676  7b274e83-1f5e-47c1-bf99-277822ea6d2e   
7899   2ac89f33-ea9f-4d6e-a74b-a32f4ba28a39   
13257  74f78ab7-9337-454c-9493-5f113ba9e1ec   
...                                     ...   
11065  101ca3bf-49b6-4a18-bf08-c565df93e562   
3761   c419f2b5-7d79-4b44-8633-ff19d95807d2   
6504   fa73e484-96c7-4dbe-aaa8-78396259a1a5   
5577   7c73accd-ffc3-4373-828d-4d64cab74ebe   
13135  d8c52567-622b-4634-9658-43ed94a7307a   

                                 Country       Product Import_Export  \
4475                      Cayman Islands  particularly        Export   
14872                              Korea          food        Export   
11676                  Wallis and Futuna         visit        Import   
7899                          Mauritania           use        Export   
13257                               Iraq           ago        Export   
...                                  ...           ...           ...   
11065                            Bahrain           nor        Export   
3761                               Chile       measure        Import   
6504   Heard Island and McDonald Islands          side        Export   
5577                          Micronesia     necessary        Import   
13135                            Romania           red        Export   

       Quantity    Value        Date     Category            Port  \
4475       3401  7596.83  21-11-2021     Clothing      Connorfort   
14872      9065  9840.43  18-08-2021     Clothing    Robinsonfort   
11676      8263  9784.94  28-01-2021  Electronics     Josephville   
7899        861  5886.57  08-08-2024  Electronics     Alberthaven   
13257      4736  5980.04  22-01-2020    Furniture      Lucasshire   
...         ...      ...         ...          ...             ...   
11065      2885  3827.47  02-02-2023         Toys  East Carolfurt   
3761       5294  9724.69  02-11-2021  Electronics  Anthonyborough   
6504       9716  8273.51  16-10-2020     Clothing     Amandaburgh   
5577       9339  1999.65  08-02-2021  Electronics       Cruzmouth   
13135      4042  3421.42  04-05-2024  Electronics     Melaniestad   

       Customs_Code   Weight Shipping_Method                        Supplier  \
4475         421451  2297.29             Air                     Wang-Pierce   
14872        273361  1355.64             Sea    Watkins, Turner and Mckenzie   
11676        595425  2794.72             Sea                     Baird-Evans   
7899         938768   651.74            Land   Herrera, Martinez and Mathews   
13257        694309  1153.97            Land                Gutierrez-Lawson   
...             ...      ...             ...                             ...   
11065        570095  4553.75             Air                     Sanders LLC   
3761         133221  2136.84            Land                     Marks-Blair   
6504         762471  4240.21             Air                     Holland Inc   
5577         261036   624.63             Sea  Garcia, Matthews and Hernandez   
13135        207281  2774.58             Air                      Parker PLC   

               Customer  Invoice_Number     Payment_Terms  
4475    Nicholas Thomas        93759184            Net 30  
14872    Rebecca Howell        51859480  Cash on Delivery  
11676       Alfred Webb        90639306            Net 60  
7899    Javier Thompson        78531183  Cash on Delivery  
13257    Anthony Warren        42585126            Net 60  
...                 ...             ...               ...  
11065     Tiffany Wolfe        40357903  Cash on Delivery  
3761       Donald Jones        40735134  Cash on Delivery  
6504     Heather Guzman        75098189           Prepaid  
5577   Christopher Lane        20721735            Net 60  
13135     Rachel Morgan        58643812  Cash on Delivery  

[2001 rows x 16 columns]

In [138]:
#Shape of the sampled data
samData.shape

(2001, 16)

In [140]:
#Viewing the information about the columns(Object Types) of the Sampled Data set
samData.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2001 entries, 4475 to 13135
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Transaction_ID   2001 non-null   object 
 1   Country          2001 non-null   object 
 2   Product          2001 non-null   object 
 3   Import_Export    2001 non-null   object 
 4   Quantity         2001 non-null   int64  
 5   Value            2001 non-null   float64
 6   Date             2001 non-null   object 
 7   Category         2001 non-null   object 
 8   Port             2001 non-null   object 
 9   Customs_Code     2001 non-null   int64  
 10  Weight           2001 non-null   float64
 11  Shipping_Method  2001 non-null   object 
 12  Supplier         2001 non-null   object 
 13  Customer         2001 non-null   object 
 14  Invoice_Number   2001 non-null   int64  
 15  Payment_Terms    2001 non-null   object 
dtypes: float64(2), int64(3), object(11)
memory usage: 265.8+ KB


In [142]:
#Data Variable Category
Index_variable = samData[['Transaction_ID', 'Date','Invoice_Number','Customs_Code']]
Non_categorical_variable = samData[['Quantity', 'Value', 'Weight']]
Categorical_nominal_variable = samData[['Country', 'Product', 'Shipping_Method', 'Supplier', 'Customer', 'Import_Export','Port', 'Category']]
Categorical_ordinal_variable = samData[['Payment_Terms']]


In [148]:
# Descriptive Statistics of the Non-Categorical Data
Non_categorical_variable.describe()

Quantity        Value       Weight
count  2001.000000  2001.000000  2001.000000
mean   4961.306347  5028.538356  2512.297186
std    2895.453319  2858.375060  1462.513339
min      11.000000   109.900000     1.520000
25%    2384.000000  2552.990000  1262.980000
50%    4946.000000  5016.560000  2475.440000
75%    7474.000000  7494.420000  3800.390000
max    9995.000000  9995.580000  4999.930000

In [164]:
# Descriptive Statistics of Categorical Data
Categorical_nominal_variable.max()

Country                                    Zimbabwe
Product                                    yourself
Shipping_Method                                 Sea
Supplier           Zimmerman, Anderson and Williams
Customer                            Zachary Summers
Import_Export                                Import
Port                                     Youngville
Category                                       Toys
dtype: object

## Project Objectives 

The objective of this project is to conduct a thorough analysis of the sample taken from the import and export data and present findings and inferences.

## Analysis and Findings from Sample Data

In [180]:
# 1. What is the total export value across all countries?

total_export_value = samData.loc[samData['Import_Export'] == 'Export', 'Value'].astype(float).sum()
print(f"Total income generated from exports across all countries: {total_export_value}")

Total income generated from exports across all countries: 4980215.92


In [770]:
#1.1 What is the total exported quantity across all countries?
total_export_quantity = samData.loc[samData['Import_Export'] == 'Export', 'Quantity'].astype(float).sum()
print(f"Total income generated from exports across all countries: {total_export_quantity}")

Total income generated from exports across all countries: 4872646.0


In [182]:
# 2. What is the total import value across all countries?
total_import_value = samData.loc[samData['Import_Export'] == 'Import', 'Value'].astype(float).sum()
print(f"Total import value across all countries: {total_import_value}")

Total import value across all countries: 5081889.33


In [776]:
#2.1 What is the total imported quantity across all countries?
total_import_quantity = samData.loc[samData['Import_Export'] == 'Import', 'Quantity'].astype(float).sum()
print(f"Total income generated from exports across all countries: {total_import_quantity}")

Total income generated from exports across all countries: 5054928.0


In [194]:
#3 Net Gain or Loss from exports and imports over the given time period.
if total_export_value>total_import_value:
    print(f"Net Income gained from trade of goods:{total_export_value-total_import_value:.2f}")
else:
    print(f"Net Loss incurred from trade of goods:{total_import_value-total_export_value:.2f}")

Net Loss incurred from trade of goods:101673.41


In [655]:
# 3. What are the months where trading peaks and dips?

samData['Date'] = pd.to_datetime(samData['Date'], errors='coerce')

# Check if conversion was successful
if samData['Date'].isnull().any():
    print("There were some errors in converting dates. Please check your data.")

samData['Year'] = samData['Date'].dt.year
samData['Month'] = samData['Date'].dt.month

# Group by Year and Month and calculate the sum of TradeVolume for each month
monthly_trade = samData.groupby(['Year', 'Month'])['Value'].sum().reset_index()

# Find the peak and dip months
peak_month = monthly_trade.loc[monthly_trade['Value'].idxmax()]
dip_month = monthly_trade.loc[monthly_trade['Value'].idxmin()]

print("Month with the highest value of trade :")
print(peak_month)

print("\nDip Month:")
print(dip_month)

Month with the highest value of trade :
Year       2020.00
Month        12.00
Value    240828.85
Name: 15, dtype: float64

Dip Month:
Year      2024.00
Month        9.00
Value    52880.45
Name: 60, dtype: float64


In [758]:
#4.1  Which countries have the highest average quantity per transaction, and how does this affect total value?
samData.groupby('Country')['Quantity'].mean().sort_values(ascending=False).head()

Country
Brazil                         8131.000000
French Southern Territories    7782.428571
Mongolia                       7777.000000
China                          7764.000000
Nicaragua                      7438.833333
Name: Quantity, dtype: float64

In [760]:
#4.2 Which country has the highest number of transactions by quantity, and what is its total transaction value?

country_agg = samData.groupby('Country').agg({
    'Quantity': 'sum',
    'Value': 'sum'
}).reset_index()

# Find the country with the highest number of transactions by quantity
max_quantity_country = country_agg.loc[country_agg['Quantity'].idxmax()]

print("Country with the highest number of transactions by quantity:")
print(max_quantity_country[['Country', 'Quantity', 'Value']])

Country with the highest number of transactions by quantity:
Country        Korea
Quantity      100901
Value       90017.51
Name: 114, dtype: object


In [822]:
# 5.Which country is the largest importer and the largest exporter by quantity and total transaction value?
exportdata = samData[samData['Import_Export']=='Export']
exportdata.groupby('Country')['Value'].sum().sort_values(ascending=False).head(3)

Country
Korea                       64008.74
Finland                     54909.86
Northern Mariana Islands    50698.75
Name: Value, dtype: float64

Korea is the country that exports the most, with a total export value of 64009.

In [824]:
importdata = samData[samData['Import_Export']=='Import']
importdata.groupby('Country')['Value'].sum().sort_values(ascending=False).head(3)

Country
Portugal        52704.69
Saint Martin    52339.96
Algeria         50247.60
Name: Value, dtype: float64

Portugal is the country that imports the most, with a total import value of 52705.

In [665]:
#6 Which countries have the highest average transaction value?
samData.groupby('Country')['Value'].mean().sort_values(ascending=False).head()

Country
Indonesia       8344.988333
Sri Lanka       8276.500000
Togo            7733.795000
Vanuatu         7306.981250
Saint Martin    7234.653750
Name: Value, dtype: float64

In [667]:
#7 What are the biggest product category that are being traded, by transaction value?
samData.groupby('Category')['Value'].sum().sort_values(ascending = False)

print("The most valuable product category is:")
print(samData.groupby('Category')['Value'].sum().sort_values(ascending = False).head(1))

The most valuable product category is:
Category
Furniture    2158053.94
Name: Value, dtype: float64


In [669]:
#8 What is the most traded product category, quantity wise?
print(samData.groupby('Category')['Quantity'].count().sort_values(ascending = False).head(1))

Category
Clothing    416
Name: Quantity, dtype: int64


In [671]:
#9 What is the average value of transaction of each product category?
samData.groupby('Category')['Value'].mean().sort_values(ascending = False)

Category
Furniture      5200.129976
Electronics    5128.039120
Clothing       4961.345938
Machinery      4932.943368
Toys           4903.529974
Name: Value, dtype: float64

In [673]:
#10 Bifurcate the most valuable product categories on the basis of export and import.
samData[samData['Import_Export'] == 'Import'].groupby(['Category','Import_Export'])['Value'].sum().sort_values(ascending = False)

Category     Import_Export
Furniture    Import           1113907.99
Clothing     Import           1081041.30
Electronics  Import            995165.47
Machinery    Import            979375.52
Toys         Import            912399.05
Name: Value, dtype: float64

The most imported product category on the basis of transaction value is Furniture.

In [676]:
samData[samData['Import_Export'] == 'Export'].groupby(['Category','Import_Export'])['Value'].sum().sort_values(ascending = False)

Category     Import_Export
Electronics  Export           1102202.53
Furniture    Export           1044145.95
Clothing     Export            982878.61
Toys         Export            955845.87
Machinery    Export            895142.96
Name: Value, dtype: float64

The most exported product category on the basis of transaction value is Electronics.

In [679]:
#11 What product category does Korea trade the most, as the country that has the biggest transaction value and the biggest exporter?
samData[samData['Country']=='Korea'].groupby('Category')['Value'].sum().sort_values(ascending = False)

Category
Electronics    31641.16
Clothing       21335.03
Furniture      20878.35
Toys           12646.11
Machinery       3516.86
Name: Value, dtype: float64

While Furniture is the category of good that is being traded the most, Korea operates the most in the category of electronics.
It can also be seen that while Korea has a decent value of trade in most categories, there is a significant drop when in comes to trade of Machineries.

In [682]:
#12  What product category does Portugal trade the most, as the country that is the biggest importer?
samData[samData['Country']=='Portugal'].groupby('Category')['Value'].sum().sort_values(ascending = False)

Category
Furniture      32281.10
Clothing       24815.79
Electronics     9376.03
Toys            1680.14
Machinery       1293.64
Name: Value, dtype: float64

Portugal trades mostly in Furniture and Clothing, with Furniture being it's biggest category. However there is a significant drop off in trade value first with Electronics, then further with Toys and Machinery.

In [685]:
#13 What is the preferred shipping method?
samData.groupby('Shipping_Method')['Country'].count().sort_values(ascending = False)


Shipping_Method
Sea     678
Land    676
Air     647
Name: Country, dtype: int64

In [687]:
#14 What is the preferred shipping method on the basis of import and export?
exportdata.groupby('Shipping_Method')['Country'].count().sort_values(ascending = False)

Shipping_Method
Sea     345
Air     340
Land    306
Name: Country, dtype: int64

In [689]:
importdata.groupby('Shipping_Method')['Country'].count().sort_values(ascending = False)

Shipping_Method
Land    370
Sea     333
Air     307
Name: Country, dtype: int64

In [691]:
#15 What is the value of goods being shipped by land, air and sea?
samData.groupby('Shipping_Method')['Value'].sum().sort_values(ascending = False)

Shipping_Method
Sea     3590442.32
Land    3339551.19
Air     3132111.74
Name: Value, dtype: float64

In [693]:
#16 For each shipping method, what is the most preferred product category?
print("For goods shipped by air, the most traded category is:\n"
      ,samData[samData['Shipping_Method']=='Air'].groupby('Category')['Value'].sum().sort_values(ascending = False).head(1))

print("\nFor goods shipped by sea, the most traded category is:\n",
      samData[samData['Shipping_Method']=='Sea'].groupby('Category')['Value'].sum().sort_values(ascending = False).head(1))

print("\nFor goods shipped by Land, the most traded category is:\n"
      ,samData[samData['Shipping_Method']=='Land'].groupby('Category')['Value'].sum().sort_values(ascending = False).head(1))


For goods shipped by air, the most traded category is:
 Category
Furniture    691579.19
Name: Value, dtype: float64

For goods shipped by sea, the most traded category is:
 Category
Electronics    781741.92
Name: Value, dtype: float64

For goods shipped by Land, the most traded category is:
 Category
Electronics    737393.56
Name: Value, dtype: float64


In [695]:
#17 What is the most valuable port?
samData.groupby('Port')['Value'].mean().sort_values(ascending = False).head(1)

Port
Brendamouth    9995.58
Name: Value, dtype: float64

In [756]:
#18 What is the port with the biggest transaction value?
samData.groupby('Port')['Value'].sum().sort_values(ascending = False).head(1)

Port
Lake Michael    28846.73
Name: Value, dtype: float64

In [699]:
#19 Which supplier has the most transactions?
samData.groupby('Supplier')['Value'].count().sort_values(ascending = False).head(1)

Supplier
Smith Ltd    4
Name: Value, dtype: int64

In [701]:
#20 What is the mean weight of trade?
samData[['Weight']].mean()

Weight    2512.297186
dtype: float64

In [703]:
#21 What is the mean weight of trade for each shipping method?
samData.groupby('Shipping_Method')['Weight'].mean().sort_values(ascending = False)

Shipping_Method
Sea     2552.361844
Air     2523.170665
Land    2461.706982
Name: Weight, dtype: float64

In [705]:
#22 What is the mean weight for each product category?
samData.groupby('Category')['Weight'].mean().sort_values(ascending = False)

Category
Toys           2575.875302
Clothing       2575.574183
Machinery      2480.891868
Furniture      2468.338627
Electronics    2462.493594
Name: Weight, dtype: float64

In [707]:
#23. Biggest Transaction in the sample data
samData.loc[samData['Value'].idxmax()]

Transaction_ID     5eefe32c-78d5-46b2-a713-facc4b88b15e
Country                                         Namibia
Product                                            land
Import_Export                                    Import
Quantity                                           2993
Value                                           9995.58
Date                                2020-07-10 00:00:00
Category                                    Electronics
Port                                        Brendamouth
Customs_Code                                     960615
Weight                                          1891.43
Shipping_Method                                    Land
Supplier                     Moore, Walker and Williams
Customer                                  Michael Brown
Invoice_Number                                 69884930
Payment_Terms                                   Prepaid
Year                                               2020
Month                                           

In [709]:
#24 Bifurcation of transactions by payment terms
samData.groupby('Payment_Terms')['Country'].count()

Payment_Terms
Cash on Delivery    536
Net 30              473
Net 60              505
Prepaid             487
Name: Country, dtype: int64

In [736]:
samData.groupby('Payment_Terms')['Value'].sum()

Payment_Terms
Cash on Delivery    2743229.24
Net 30              2441414.71
Net 60              2379178.84
Prepaid             2498282.46
Name: Value, dtype: float64

In [738]:
samData.groupby('Payment_Terms')['Value'].mean()

Payment_Terms
Cash on Delivery    5117.965000
Net 30              5161.553298
Net 60              4711.245228
Prepaid             5129.943450
Name: Value, dtype: float64

In [764]:
#25 Is there a statistically significant average Transaction Value of import and export.
from scipy import stats
import_data = samData[samData['Import_Export'] == 'Import']['Value']
export_data = samData[samData['Import_Export'] == 'Export']['Value']
t_stat, p_value = stats.ttest_ind(import_data, export_data, equal_var=False)  # equal_var=False for Welch's T-test (more robust for unequal variances)

print("T-Test Results between Import and Export Categories")
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")


if p_value < 0.05:
    print("There is a statistically significant difference in average Transaction Value between Import and Export categories.")
else:
    print("There is no statistically significant difference in average Transaction Value between Import and Export categories.")

T-Test Results between Import and Export Categories
T-statistic: 0.047936664986846775
P-value: 0.9617715113704851
There is no statistically significant difference in average Transaction Value between Import and Export categories.


In [742]:
#26 Is there a signifcant difference in average Transaction Value across the shipping methods?

shipping_air = samData[samData['Shipping_Method'] == 'Air']['Value']
shipping_sea = samData[samData['Shipping_Method'] == 'Sea']['Value']
shipping_land = samData[samData['Shipping_Method'] == 'Land']['Value']

f_statistic, p_value = stats.f_oneway(shipping_air, shipping_sea, shipping_land)
print("ANOVA Test Results for Shipping Methods")
print(f"F-statistic: {f_statistic}")
print(f"P-value: {p_value}")


if p_value < 0.05:
    print("There is a statistically significant difference in average Transaction Value across the three shipping methods.")
else:
    print("There is no statistically significant difference in average Transaction Value across the three shipping methods.")

ANOVA Test Results for Shipping Methods
F-statistic: 4.693382032898953
P-value: 0.009256849638839918
There is a statistically significant difference in average Transaction Value across the three shipping methods.


In [790]:
#27 Is there a significant difference 
shipping_methods = samData['Shipping_Method'].unique()
weights_by_method = [samData[samData['Shipping_Method'] == method]['Weight'] for method in shipping_methods]
f_statistic, p_value = stats.f_oneway(*weights_by_method)
print("ANOVA Test Results")
print(f"F-statistic: {f_statistic}")
print(f"P-value: {p_value}")
if p_value < 0.05:
    print("There is a statistically significant difference in weight across different shipping methods.")
else:
    print("There is no statistically significant difference in weight across different shipping methods.")

ANOVA Test Results
F-statistic: 0.6765023653090835
P-value: 0.5085084674151488
There is no statistically significant difference in weight across different shipping methods.


In [817]:
#28 Is there a significant difference between weights of transaction in different shipping methods?
weight_summary = samData.groupby('Shipping_Method')['Weight'].agg(['mean', 'sum']).reset_index()
weight_summary.columns = ['Shipping_Method', 'Average_Weight', 'Total_Weight']


print("Weight Summary Statistics by Shipping Method:")
print(weight_summary)

shipping_methods = samData['Shipping_Method'].unique()

# Create a list of weight data for each shipping method
weights_by_method = [samData[samData['Shipping_Method'] == method]['Weight'] for method in shipping_methods]

# Step 3: Perform ANOVA Test
f_statistic, p_value = stats.f_oneway(*weights_by_method)

# Print ANOVA results
print("\nANOVA Test Results")
print(f"F-statistic: {f_statistic}")
print(f"P-value: {p_value}")

# Interpretation of the ANOVA result
if p_value < 0.05:
    print("There is a statistically significant difference in weight across different shipping methods.")
else:
    print("There is no statistically significant difference in weight across different shipping methods.")

Weight Summary Statistics by Shipping Method:
  Shipping_Method  Average_Weight  Total_Weight
0             Air     2523.170665    1632491.42
1            Land     2461.706982    1664113.92
2             Sea     2552.361844    1730501.33

ANOVA Test Results
F-statistic: 0.6765023653090835
P-value: 0.5085084674151488
There is no statistically significant difference in weight across different shipping methods.


## Managerial Insights

1. There is currently a net loss due to the higher value of imports over exports, managerial focus should be on gaining a stronghold and converting it into a profit by increasing exports.
2. The countries to focus on as big players are Korea, Portugal, Finland, and Algeria.
3. There is a significant difference in the countries that are the absolute trade leaders and the countries with a higher average transaction value, those countries should be targeted and focused on.
4. Our Most valuable product category is Furniture, based on both, absolute and average transaction values. This can be converted into a cash cow by further improving quantity and gaining a higher market share.
5. Portugal trades mostly in Furniture and Clothing, with Furniture being its biggest category. However there is a significant drop off in trade value first with Electronics, then further with Toys and Machinery.
6. Furniture is the category of goods that is being traded the most, Korea operates the most in the category of electronics. It can also be seen that while Korea has a decent trade value in most categories, there is a significant drop in trade of machinery.
7. There is a statistically significant difference in average Transaction Value across the three shipping methods, where the transaction through "Sea" is the maximum and the transaction through "Air" being the lowest.